### PR TITLE
[terra-collapsible-menu-view] add isStartAligned prop to collapsible-menu-view

### LINKED ISSUE #1614: [terra-collapsible-menu-view] add start alignment
# Feature Request

## Description
Add a new prop isStartAligned to allow for left alignment of the menu.

## Additional Context/Screenshots
Screenshot 2023-03-20 at 4 36 31 PM

## @ Mentions
@neilpfeiffer

### PR DESCRIPTION
**Pull request by kschuste**
### Summary
This PR will add the isStartAligned prop to the terra-collapsible-menu-view.  This will allow for left alignment vs it always being right aligned.

Closes #1614

### Deployment Link

https://terra-framework-deployed-pr-#.herokuapp.com/

### Testing
I have tested these changes locally via the tests, examples, and my local code.

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by supreethmr** [comment]
Are these changes tested on IE 11 browser as well ?? just to ensure flex-start works fine without any issue

File: packages/terra-collapsible-menu-view/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Added
+  * Added isStartAligned prop
```
Comment (supreethmr) [inline packages/terra-collapsible-menu-view/CHANGELOG.md:6]:
```suggestion
  * Added `isStartAligned` prop
```

File: packages/terra-collapsible-menu-view/src/CollapsibleMenuView.jsx [hunk 2]
```diff
@@ -143,15 +149,18 @@ class CollapsibleMenuView extends React.Component {
...
+      children, boundingRef, menuWidth, intl, alwaysCollapsedMenuItems, isStartAligned, ...customProps
     } = this.props;
     const theme = this.context;
 
     const collapsibleMenuViewClassName = classNames(cx(
       'collapsible-menu-view',
       { 'is-calculating': this.isCalculating },
       theme.className,
+      { 'collapsible-menu-view-flex-end': !isStartAligned },
+      { 'collapsible-menu-view-flex-start': isStartAligned },
```
Comment (supreethmr) [inline packages/terra-collapsible-menu-view/src/CollapsibleMenuView.jsx:161]: nit: 
```suggestion
      { 'collapsible-menu-view-flex-end': !isStartAligned },
      { 'collapsible-menu-view-flex-start': isStartAligned },
      theme.className,
```

**Comment by kschuste** [comment]
> Are these changes tested on IE 11 browser as well ?? just to ensure flex-start works fine without any issue
@supreethmr 
I don't have a windows machine to test IE 11, is there a way I can test it on IE 11 with a macbook?

**Comment by kschuste** [comment]
@supreethmr can confirm that IE 11 works for the flex-start alignment
Screenshot 2023-03-29 at 12 45 51 PM

**Review by supreethmr — APPROVED** [review]

**Review by neilpfeiffer — APPROVED** [review]
Tested: PR #1613
Resolves: Issue #1614

\
**Reviewed:**
- Adds new bool prop for `isStartAligned` to be used in special uses like toolbars and clinical detail panels when actions align to the left of the container. 


\
**Missing / Fix:**
- JEST snapshots don't appear to be updated but I would have expected them to include the new prop.



 **UX Reviewed** |
-----------|

**Comment by neilpfeiffer** [comment]
All checks complete and passing. Merging as approved.

**Pull request merged**